web/src/components/apps/Dashboard.tsx

```diff
@@ -26,47 +26,136 @@ const COMMON_ERRORS = {
   "no such host": "No such host",
 };
 
-class Dashboard extends Component {
-  state = {
```

Comment:
moved this into the constructor. took me hours to figure out why some of my typechecks were giving me weird errors. it was because the initial state was getting set outside of a constructor + parts of the state were getting set in a timeout. I don't think TS could see that the state was changing inside of that timeout so it was always just falling back to the types set here.